File: Samples/iOS/iOS/Person+Request.m
//
//  Person+Request.m
//  CoreDataClientSample
//
//  Created by Nathaniel Potter on 11/4/14.
//  Copyright (c) 2014 Nathaniel Potter. All rights reserved.
//

#import "Person+Request.h"

@implementation Person (Request)

+ (void)initialize
{
    [self setBasePath:@"https://www.googleapis.com/plus/v1/people/"];
}

+ (void)personWithId:(NSString *)uid completion:(void(^)(Person *person, NSError *error))completion
{
    [self GET:uid progress:nil request:nil withCompletion:^(NSArray *results, NSError *error) {
        completion(results.lastObject, error);
    }];
}

@end

File: Samples/iOS_Swift/CoreDataClientSample/Person+Request.swift
//
//  Person+Request.swift
//  CoreDataClientSample
//
//  Created by Nathaniel Potter on 1/31/15.
//  Copyright (c) 2015 Nathaniel Potter. All rights reserved.
//

import Foundation

extension Person {
    
    struct Initialize {
        static var basePath: String = "https://www.googleapis.com/plus/v1/people/"
        static var managedObjectUidKey: String = "id"
        static var responseObjectUidKey: String = "id"
    }
    
    class var basePath: String {
        return Initialize.basePath
    }
    
    class var managedObjectUidKey: String {
        return Initialize.managedObjectUidKey
    }
    
    class var responseObjectUidKey: String {
        return Initialize.responseObjectUidKey
    }
    
    class func personWithId(id: String, completion:(person: Person?, error: NSError?)->()) {
        self.GET(id, withCompletion: { (results: Array!, error: NSError!) -> Void in
            if (results.count > 0) {
                let person = results[0] as Person
                completion(person: person, error: error)
            } else {
                completion(person: nil, error: error)
            }
        })
    }
}
